### ISSUE TITLE
usage to stdout

### ISSUE THREAD

**Issue by glensc** [lxc/lxd#2301]
consider using `stdout` instead of `stderr` printing usage information so the lengthy output could be viewed with for example `| less`:

```
# lxc config | less
```

**Comment by stgraber** [comment]
So I'd actually expect "lxc config" to go to stderr as it's printing the help as a result of an error "wrong number of subcommand arguments".

However, "lxc help" or "lxc config --help" should be going to stdout as the user actually did request the help.

It looks like we're not doing things quite that way right now, as --help and the help command both go to stderr when they shouldn't.

**Comment by glensc** [comment]
agree with errors to stderr part, but the problem is that "lxc config" is not just showing error but also dumping usage (which is the long part). it probably should just show error and suggest to use "lxc config --help"

i think that's the way to go, as it's really hard to notice errors if some long output is printed (that usage help)

**Comment by stgraber** [comment]
We always dump usage on command parsing error in the lxc tool as a convenience. Maybe we should stop doing that, but it's been pretty useful to our users I think.

We used to print the error at the top which was particularly hard to find, though @tych0 moved the error message to the bottom a little while back to make it easier for folks to see.

**Comment by tych0** [comment]
On Sun, Aug 21, 2016 at 08:15:28AM -0700, Stéphane Graber wrote:
> We always dump usage on command parsing error in the lxc tool as a convenience. Maybe we should stop doing that, but it's been pretty useful to our users I think.
> 
> We used to print the error at the top which was particularly hard to find, though @tych0 moved the error message to the bottom a little while back to make it easier for folks to see.

We could try to colorize the errors, perhaps. Would that help?

**Comment by glensc** [comment]
I think errors should be brief and concise. `lxc config --help` could be added as single line aside the error message to assist new users. for advanced users always showing usage aside error makes errors hard to follow and makes more efforts to eyes to extract error from usage noise.
